test/xds/xds_client_ack_nack_test.go

```diff
@@ -65,8 +66,19 @@ func (s) TestClientResourceVersionAfterStreamRestart(t *testing.T) {
 	}
 	lis := testutils.NewRestartableListener(l)
 
-	streamClosed := grpcsync.NewEvent() // Event to notify stream closure.
-	acksReceived := grpcsync.NewEvent() // Event to notify receipt of acks for all resources.
+	// We depend on the fact that the management server assigns monotonically
+	// increasing stream IDs starting at 1.
+	const (
+		idBeforeRestart = 1
+		idAfterRestart  = 2
+	)
+
+	// Events of importance in the test, in the order in which they are expected
+	// to happen.
+	acksReceivedBeforeRestart := grpcsync.NewEvent()
+	streamRestarted := grpcsync.NewEvent()
```

Comment:
Nit: why did you rename this from streamClosed to streamRestarted? streamClosed feels more appropriate.

Reply:
I wanted to make it extra clear that the stream actually *restarts*, and that it is not a permanent stream closure. Let me know if you feel strongly though, and I can switch back.